xray/utils/auditbasicparams.go

```diff
@@ -18,6 +18,8 @@ type AuditBasicParams struct {
 	args                    []string
 	depsRepo                string
 	ignoreConfigFile        bool
+	npmIgnoreNodeModules    bool
+	npmOverWritePackageLock bool
```

Comment:
I'm not sure it is a good idea to have npm specific arguments in a generic audit struct.
Can we use inheritance? Something like:
```go
type AuditNpmParams struct {
	AuditBasicParams
	npmIgnoreNodeModules    bool
	npmOverWritePackageLock bool
}
```

Reply:
we already have specific arguments for python in the "generic" audit struct (was there before I did my changes)
I thought that once I will support more pkg type, I will do something more generic

Reply:
I've pushed changes with refactor, please cheeck it out 🙏